### PR TITLE
MC-23989: Mini cart missing when you edit inline welcome message for guest and use special characters

### LINKED ISSUE #32250: Mini cart missing when you edit inline welcome message for guest and use special characters
**Note:** Bug was found on Magento installed from composer on <https://m233-se-dev-0.testregress.com/>

**Precondition**
1. Developer mode is set
1. Enable Translate Inline functionality
 Go to Stores -> Configuration -> Advanced -> Developer -> Translate Inline: 
 Enabled for Storefront - Yes 
 Save Config
1. php bin/magento cache:flush
1. Create simple product

**Steps to reproduce**
1. Go to storefront as guest and add product to cart - product added, you can see mini cart
1. Edit inline *Default welcome message_, change value to: _Welcome to "Food & Drinks" store*
1. In another tab Go to Admin -> Stores -> Configuration -> Advanced -> Developer -> Translate Inline:
 Enabled for Storefront - No 
 Save Config
1. php bin/magento cache:flush
1. Go back to storefront and refresh page

**Expected result:**
 (/) New welcome message correctly displayed. 
 (/) Mini cart with product inside displayed. 
 (/) Adding to cart one more product performed with success message and displayed in mini shopping cart.
 (/) No errors in browser console.

**{color:#d04437}Actual result:**
 (x) Welcome message is missing. 
 (x) Mini cart with product inside not displayed. 
 (x) Adding to cart one more product performed without success message and not displayed in mini shopping cart.
 (x) Errors in browser console.
 !Screenshot from 2021-02-22 12-55-55.png|width=926,height=324!

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

Fixes: https://github.com/magento/magento2/issues/32250

### Manual testing scenarios (*)

1. See: https://github.com/magento/magento2/issues/32250

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Theme/view/frontend/templates/html/header.phtml [hunk 1]
```diff
@@ -6,19 +6,20 @@
...
         <!-- ko if: customer().fullname  -->
         <span class="logged-in"
-              data-bind="text: new String('<?= $block->escapeHtml(__('Welcome, %1!', '%1')) ?>').replace('%1', customer().fullname)">
+              data-bind="text: new String('<?= $escaper->escapeHtml(__('Welcome, %1!', '%1')) ?>').replace('%1', customer().fullname)">
         </span>
         <!-- /ko -->
         <!-- ko ifnot: customer().fullname  -->
         <span class="not-logged-in"
-              data-bind='html:"<?= $block->escapeHtml($welcomeMessage) ?>"'></span>
+              data-bind="html: "<?= $escaper->escapeHtml($welcomeMessage) ?>"></span>
```
Comment (zakdma) [inline app/code/Magento/Theme/view/frontend/templates/html/header.phtml]: This solution looks unreliable. Please tru to leave quotes as was but change escaper method to escapeHtmlAttribute
Please add tests to cover changes

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by zakdma** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests EE

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32251`.

**Comment by zakdma** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento run Functional Tests B2B

**Comment by zakdma** [comment]
@magento run Functional Tests B2B

**Comment by zakdma** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32251`.

**Pull request merged**